src/merkle/simple_smt/mod.rs

```diff
@@ -9,13 +9,20 @@ mod tests;
 // ================================================================================================
 
 /// A sparse Merkle tree with 64-bit keys and 4-element leaf values, without compaction.
-/// Manipulation and retrieval of leaves and internal nodes is provided by its internal `Store`.
 /// The root of the tree is recomputed on each new leaf update.
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct SimpleSmt {
     root: Word,
+    leaves: BTreeMap<u64, Word>,
     depth: u8,
-    pub(crate) store: Store,
+    empty_hashes: Vec<RpoDigest>,
+    pub(crate) branches: BTreeMap<NodeIndex, BranchNode>,
+}
```

Comment:
minor nit: I'd probably change the order here to be like:
```Rust
pub struct SimpleSmt {
    depth: u8,
    root: Word,
    leaves: BTreeMap<u64, Word>,
    pub(crate) branches: BTreeMap<NodeIndex, BranchNode>,
    empty_hashes: Vec<RpoDigest>,
}
```
Also, would be nice to make `branches` private - but maybe in a different PR.

Reply:
changed fields order.

making the `branches` private requires a different way of getting to the nodes, something like https://github.com/0xPolygonMiden/crypto/pull/117 would do. (I started this cleanup as I was reading the code to figure out how much work would that be).

I agree it should be done on separate PR though :)